C compilation error on doing `return {}`, when the return type of a function is `?map[string]string`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.3 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/cleanv
vmodules: /home/delian/.vmodules
vroot: /v/cleanv
vexe: /v/cleanv/v
vexe mtime: 2022-01-09 18:55:00
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 214853c.5e85d4c

Git version: git version 2.25.1
Git vroot status: weekly.2022.01-21-g5e85d4cb
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg a.v`
```v
fn opt() ?map[string]string {
	return {}
}

fn main() {
	x := opt() ?
	dump(x)
}

```

**What did you expect to see?**

working and running program


**What did you see instead?**
```
==================
/tmp/v_1000/a.12731624042494696098.tmp.c:10993: error: '{' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/containers.v
+++ b/vlib/v/checker/containers.v
@@ -202,7 +202,7 @@ pub fn (mut c Checker) map_init(mut node ast.MapInit) ast.Type {
 		sym := c.table.sym(c.expected_type)
 		if sym.kind == .map {
 			info := sym.map_info()
-			node.typ = c.expected_type
+			node.typ = c.expected_type.clear_flag(.optional)
 			node.key_type = info.key_type
 			node.value_type = info.value_type
 			return node.typ

--- a/vlib/v/tests/option_empty_map_test.v
+++ b/vlib/v/tests/option_empty_map_test.v
@@ -1,0 +1,13 @@
+fn opt() ?map[string]string {
+	return {}
+}
+
+fn test_option_empty_map() {
+	x := opt() or {
+		assert false
+		return
+	}
+
+	dump(x)
+	assert '$x' == '{}'
+}

